block_device.go

```diff
@@ -45,6 +53,32 @@ func (bd *BlockDevice) Write(p []byte) (int, error) {
 			return 0, err
 		}
 
+		// From <mtd/ubi-user.h>
+		//
+		// UBI volume update
+		// ~~~~~~~~~~~~~~~~~
+		//
+		// Volume update should be done via the UBI_IOCVOLUP ioctl command of the
+		// corresponding UBI volume character device. A pointer to a 64-bit update
+		// size should be passed to the ioctl. After this, UBI expects user to write
+		// this number of bytes to the volume character device. The update is finished
+		// when the claimed number of bytes is passed. So, the volume update sequence
+		// is something like:
+		//
+		// fd = open("/dev/my_volume");
+		// ioctl(fd, UBI_IOCVOLUP, &image_size);
+		// write(fd, buf, image_size);
+		// close(fd);
+		if bd.typeUBI {
+			_, _, err := syscall.Syscall(syscall.SYS_IOCTL, out.Fd(),
+				uintptr(unsafe.Pointer(UBI_IOCVOLUP)),
+				uintptr(unsafe.Pointer(&bd.ImageSize)))
+			if err != 0 {
+				log.Errorf("Failed to write images size to UBI_IOCVOLUP: %v", err)
+				return 0, err
+			}
+		}
```

Comment:
Could you move this syscall also to a separate function in `ioctl.go`? The reason is that this is very Linux specific, and we are considering ports to other platforms such as FreeBSD. It's easier if things like these are collected in one place.